### PR TITLE
Issue/8326 page upload notification

### LINKED ISSUE #8326: Make upload notifications compatible with PPP
After a page is uploaded while the app is in the background, a notification is displayed. When users tap the notification they are taken to the old page list. This needs to be modified to work with the new Pages.

### PR DESCRIPTION
**Pull request by 0nko**
Fixes #8326.

To test:
1. Open a site that has many pages (so you have to scroll the list)
2. Open a page and edit it
3. Quickly exit the Site pages before the upload is finished (otherwise the notification isn't shown)
4. When the upload is done, a notification is shown
5. Tap the notification
6. Notice the Site pages screen is opened, scrolled so that the changed page is visible on the screen

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/pages/PageListFragment.kt [hunk 1]
```diff
@@ -103,4 +106,34 @@ class PageListFragment : Fragment() {
         }
         adapter.update(pages)
     }
+
+    fun scrollToPage(pageId: Long) {
+        val position = viewModel.pages.value?.indexOfFirst { it is Page && it.id == pageId }
+        if (position != null && position != -1) {
```
Comment (malinajirka) [inline WordPress/src/main/java/org/wordpress/android/ui/pages/PageListFragment.kt]: Do we really want to silently fail if the position is invalid? I'd consider either throwing an exception or logging a message. Wdyt?

File: WordPress/src/main/java/org/wordpress/android/ui/pages/PageListFragment.kt [hunk 2]
```diff
@@ -103,4 +106,34 @@ class PageListFragment : Fragment() {
...
+            val smoothScroller = object : LinearSmoothScroller(activity) {
+                private val SCROLL_OFFSET_DP = 23
```
Comment (malinajirka) [inline WordPress/src/main/java/org/wordpress/android/ui/pages/PageListFragment.kt]: We may want to consider moving this value into dimens.xml. Wdyt?

File: WordPress/src/main/java/org/wordpress/android/ui/pages/PagesActivity.kt [hunk 3]
```diff
@@ -25,6 +26,19 @@ class PagesActivity : AppCompatActivity(), BasicDialogPositiveClickInterface, Ba
         }
     }
 
+    override fun onNewIntent(intent: Intent) {
+        super.onNewIntent(intent)
+
+        if (intent.hasExtra(EXTRA_PAGE_REMOTE_ID_KEY)) {
+            val pageId = intent.getLongExtra(EXTRA_PAGE_REMOTE_ID_KEY, -1)
+            if (pageId != -1L) {
```
Comment (malinajirka) [inline WordPress/src/main/java/org/wordpress/android/ui/pages/PagesActivity.kt]: We may want to consider throwing an exception, wdyt?

File: WordPress/src/main/java/org/wordpress/android/ui/pages/PageListFragment.kt (hunk 1 above)
Comment (0nko) [reply WordPress/src/main/java/org/wordpress/android/ui/pages/PageListFragment.kt]: If a page is missing then we don't want to crash the app. It means the page might have been deleted or something. So all that will happen is that the Pages activity is displayed without the scrolling. I think failing silently is ok here.

Comment (malinajirka) [reply WordPress/src/main/java/org/wordpress/android/ui/pages/PageListFragment.kt]: 👍

**Comment by malinajirka** [comment]
The navigation works as expected, however the scrollToPage still doesn't work:(.

upload-notif

**Pull request merged**